anxcloud/data_source_tags.go

```diff
@@ -23,17 +23,22 @@ func dataSourceTagsRead(ctx context.Context, d *schema.ResourceData, m interface
 	c := m.(providerContext).legacyClient
 	tagsAPI := tags.NewAPI(c)
 
-	page := d.Get("page").(int)
-	limit := d.Get("limit").(int)
 	query := d.Get("query").(string)
 	serviceIdentifier := d.Get("service_identifier").(string)
 	organizationIdentifier := d.Get("organization_identifier").(string)
 	order := d.Get("order").(string)
 	sortAscending := d.Get("sort_ascending").(bool)
 
-	tags, err := tagsAPI.List(ctx, page, limit, query, serviceIdentifier, organizationIdentifier, order, sortAscending)
-	if err != nil {
-		return diag.FromErr(err)
+	var tags []tags.Summary
+	for page := 1; page < 10; page++ { // limit to 10K tags (should be more than enough)
```

Comment:
While this limit is fine in pretty much all cases, it might not be in every case and with removing pagination from the interface, there is no way a user could retrieve the next batch of 10k tags

Why have a limit at all here?

Reply:
Yes, this shouldn't be necessary. Removed it.